internal/systemd.go

```diff
@@ -0,0 +1,20 @@
+package internal
+
+import (
+	"os/exec"
+	"strings"
+)
+
+// IsSystemShutdown tries to determine if systemd shutdown or reboot is being executed
+func IsSystemShutdown() bool {
+	return FileExists("/run/nologin") || FileExists("/var/run/nologin")
```

Comment:
it seams there is also a `/etc/nologin` to block the logins. Should it be also that checked?
It was introduced https://github.com/linux-pam/linux-pam/commit/e9e593f6ddeaf975b7fe8446d184e6bc387d450b. 

https://man7.org/linux/man-pages/man8/pam_nologin.8.html

Reply:
Changed `shutdown` detection method to `dbus` monitor for a certain system event

Reply:
yes, but I saw that the files are still checked for non-dbus.

Reply:
Final logic is to check only existence of `/run/nologin` file according to this source: https://www.freedesktop.org/software/systemd/man/latest/shutdown.html
Other files `/etc/nologin` and `/var/run/nologin` looks like have administrative purpose for sys admins to disable login to the system (as described in `pam` subsystem in your reference).